lib/broccoli/build-translation-tree.js

```diff
@@ -28,8 +29,14 @@ function buildTranslationTree(project, inputPath, treeGenerator) {
   );
 }
 
+function isEngineWithIntl(addon) {
+  return isEngine(addon) && addon.pkg.dependencies['ember-intl'];
+}
+
 function processAddons(addons, translationTrees, treeGenerator) {
-  addons.forEach(addon => _processAddon(addon, translationTrees, treeGenerator));
+  addons
+    .filter(addon => !isEngineWithIntl(addon))
```

Comment:
This line ended up being the reason for the custom treeForTranslations hook not firing.  I'll need to think about how we can preserve that behavior along with your changes.

Reply:
Is the use case for the engine to add additional paths for its own build, or to add translations to the host application (that embeds the engine)?

Reply:
The hook is used to programmatically provide translations to the engine or app.  In the context of your changes, this would be scoped to the immediate engine and not the host app.

It turned out to be a trivial change to what you've already done: https://github.com/ember-intl/ember-intl/commit/8f30184e2607ce2de92d115603eacefa30edd233

Feel free to cherry-pick into your fork & branch.

Reply:
There's still the issue where I'm finding the engine's translations being bundled as the host translations.  I'll try and poke around to see if I can figure that one out.